tests/unit/dynamic_rendering_local_read.cpp

```diff
@@ -0,0 +1,1072 @@
+    AddRequiredFeature(vkt::Feature::dynamicRenderingLocalRead);
+    SetTargetApiVersion(VK_API_VERSION_1_2);
+    RETURN_IF_SKIP(Init());
+
+    CreatePipelineHelper pipe(*this);
+    pipe.InitState();
+    pipe.CreateGraphicsPipeline();
+
+    VkRenderingAttachmentInfoKHR color_attachment = vku::InitStructHelper();
+    color_attachment.imageLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+
+    m_commandBuffer->begin();
+    vk::CmdBindPipeline(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_GRAPHICS, pipe.Handle());
+
+    VkRenderingInfo rendering_info = vku::InitStructHelper();
+    rendering_info.renderArea = {{0, 0}, {m_width, m_height}};
+    rendering_info.layerCount = 1;
+    rendering_info.colorAttachmentCount = 1;
+    rendering_info.pColorAttachments = &color_attachment;
+
+    m_commandBuffer->BeginRendering(rendering_info);
+
+    uint32_t location = VK_ATTACHMENT_UNUSED;
+    VkRenderingAttachmentLocationInfoKHR location_info = {VK_STRUCTURE_TYPE_RENDERING_ATTACHMENT_LOCATION_INFO_KHR, nullptr, 1,
+                                                          &location};
+    vk::CmdSetRenderingAttachmentLocationsKHR(*m_commandBuffer, &location_info);
+
+    VkClearAttachment clear_attachment;
+    clear_attachment.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+    clear_attachment.clearValue.color.float32[0] = 1.0;
+    clear_attachment.clearValue.color.float32[1] = 1.0;
+    clear_attachment.clearValue.color.float32[2] = 1.0;
+    clear_attachment.clearValue.color.float32[3] = 1.0;
+    clear_attachment.colorAttachment = 0;
+    VkClearRect clear_rect = {{{0, 0}, {m_width, m_height}}, 0, 1};
+
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-vkCmdClearAttachments-colorAttachment-09503");
+    vk::CmdClearAttachments(*m_commandBuffer, 1, &clear_attachment, 1, &clear_rect);
+
+    m_errorMonitor->VerifyFound();
+}
+
+TEST_F(NegativeDynamicRenderingLocalRead, ImageBarrier) {
+    TEST_DESCRIPTION("Test setting image memory barrier without dynamic rendering local read features enabled.");
+    SetTargetApiVersion(VK_API_VERSION_1_3);
+    //AddRequiredExtensions(VK_EXT_SHADER_TILE_IMAGE_EXTENSION_NAME);
+    AddRequiredExtensions(VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME);
+    AddRequiredExtensions(VK_KHR_DYNAMIC_RENDERING_LOCAL_READ_EXTENSION_NAME);
+    AddRequiredFeature(vkt::Feature::synchronization2);
+    AddRequiredFeature(vkt::Feature::dynamicRendering);
+    AddRequiredFeature(vkt::Feature::shaderTileImageColorReadAccess);
+    m_errorMonitor->SetAllowedFailureMsg("VUID-VkDeviceCreateInfo-pNext-pNext");
+    RETURN_IF_SKIP(InitBasicDynamicRendering());
+    InitRenderTarget();
+
+    m_commandBuffer->begin();
+
+    VkRenderingInfoKHR begin_rendering_info = vku::InitStructHelper();
+    VkClearRect clear_rect = {{{0, 0}, {m_width, m_height}}, 0, 1};
+    begin_rendering_info.renderArea = clear_rect.rect;
+    begin_rendering_info.layerCount = 1;
+
+    VkImageObj image(m_device);
```

Comment:
We **finally** got rid of the old `VkImageObj`

this can simply be re-written to

```suggestion
    vkt::Image image(*m_device, 32, 32, 1, VK_FORMAT_R8G8B8A8_UNORM, VK_IMAGE_USAGE_STORAGE_BIT);
```

and will work the same

Reply:
It looks in this PR I have no such constructor for vkt::Image.
What is the process? Should I rebase or merge main?

Reply:
well you need to rebase because there are already merge conflicts, once you rebase, you will have it

Reply:
Rebased and force-pushed.